### PR TITLE
Frontend development setup

### PR DESCRIPTION
**Pull request by pavish**
Related to #59 

## This PR contains:
 - Frontend setup with Svelte.
 - Tooling using Vite with hot module replacement.
 - Dev setup with Django templates.
 - Eslint rules for linting.

## Yet to handle:
- Github workflow action for UI.
- Production build setup (discussion required).
- Testing library integration.
- Documentation update.

## Details:
- Separate UI HMR service starts at port :3000 for development (Does not have to be directly accessed. Not required for production).
- Django runs at :8000 as usual. Any change in svelte code will be reflected immediately without the need for page reload.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: mathesar/templates/mathesar/base.html [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
+      <a href="/">Mathesar</a>
+    </div>
+    {% block body %}
+    {% endblock %}
+    {% block scripts %}
+    {% endblock %}
+
+    <!-- For development with hot module replacement -->
+    <!-- TODO: Control mode with an env variable -->
+    <script type="module" src="http://localhost:3000/@vite/client"></script>
```
Comment (kgodey) [inline mathesar/templates/mathesar/base.html:20]: I don't think we should hardcode this, can we use a Django setting to specify the URL here?

**Review by kgodey — CHANGES_REQUESTED** [review]
Looks pretty good to me, one minor change requested.

I haven't tried running this locally.

**Review by mathemancer — COMMENTED** [review]
Looks good, aside from what @kgodey mentioned.

File: config/settings.py [hunk 2]
```diff
@@ -133,3 +134,7 @@
 # https://docs.djangoproject.com/en/3.1/howto/static-files/
 
 STATIC_URL = "/static/"
+
+CLIENT_DEV_URL = "http://localhost:3000"
+
+DEVELOPMENT_MODE = True
```
Comment (kgodey) [inline config/settings.py:140]: there's already a Django setting for this, `settings.DEBUG`, I recommend just using it.

File: mathesar/templates/mathesar/base.html (hunk 1 above)
Comment (pavish) [reply mathesar/templates/mathesar/base.html]: I've used a context processor to be able to get the url in templates. Based on the quick reading I did, this seemed like the optimal way, but let me know if this is not okay.

Comment (kgodey) [reply mathesar/templates/mathesar/base.html]: That works.

**Review by kgodey — APPROVED** [review]

**Pull request merged**